### ISSUE TITLE
[admin] Make opentelemetrybot push tags for us

### ISSUE THREAD

**Issue by mx-psi** [open-telemetry/opentelemetry-collector#7630]
To further reduce the manual work needed during releases, we want to explore automating pushing the tags from the release job.

@codeboten assigning this to you since you are already working on it IIUC.

Relates to #7576.

**Comment by Aneurysm9** [comment]
I don't believe we can use the bot for this as it is [not supposed to have write permission](https://github.com/open-telemetry/community/blob/main/assets.md#opentelemetry-bot) to any repo.

Beyond any policy limitations with using the bot for this purpose, I have reservations around removing the requirement that the release manager sign the tags with a personal GPG key.  I get that I'm probably in the minority in signing commits and tags with a key from an HSM, but I do find it valuable to know that a particular individual has reviewed and signed off on the release and worry about the potential for the automated creation of unauthorized releases that are indistinguishable from authorized releases.
